packages/medusa/src/utils/middlewares/authenticate-middleware.ts

```diff
@@ -36,63 +36,65 @@ export const authenticate = (
     next: NextFunction
   ): Promise<void> => {
     const authTypes = Array.isArray(authType) ? authType : [authType]
+    const scopes = Array.isArray(authScope) ? authScope : [authScope]
```

Comment:
```suggestion
    const scopes = Array.isArray(authScope) ? authScope : [authScope]
    const req_ = req as AuthenticatedMedusaRequest
```

then you can use `req_` without having to cast all the time 👍